src/VisualStudio/Core/Def/ProjectSystem/VisualStudioProject.cs

```diff
@@ -239,18 +234,29 @@ private void ChangeProjectProperty<T>(ref T field, T newValue, Action<SolutionCh
 
                 field = newValue;
 
-                // Importantly, we do not await/wait on the fullyLoadedStateTask.  We do not want to ever be waiting on work
-                // that may end up touching the UI thread (As we can deadlock if GetTagsSynchronous waits on us).  Instead,
-                // we only check if the Task is completed.  Prior to that we will assume we are still loading.  Once this
-                // task is completed, we know that the WaitUntilFullyLoadedAsync call will have actually finished and we're
-                // fully loaded.
-                var isFullyLoadedTask = _workspaceStatusService?.IsFullyLoadedAsync(CancellationToken.None);
-                var isFullyLoaded = isFullyLoadedTask is { IsCompleted: true } && isFullyLoadedTask.GetAwaiter().GetResult();
-
-                // We only log telemetry during solution open
-                if (logThrowAwayTelemetry && _telemetryService?.HasActiveSession == true && !isFullyLoaded)
+                if (logThrowAwayTelemetry)
                 {
-                    TryReportCompilationThrownAway(_workspace.CurrentSolution.State, Id);
+                    var telemetryService = _workspace.Services.GetService<IWorkspaceTelemetryService>();
+
+                    if (telemetryService?.HasActiveSession == true)
+                    {
+                        var workspaceStatusService = _workspace.Services.GetService<IWorkspaceStatusService>();
+
+                        // We only log telemetry during solution open
+
+                        // Importantly, we do not await/wait on the fullyLoadedStateTask.  We do not want to ever be waiting on work
+                        // that may end up touching the UI thread (As we can deadlock if GetTagsSynchronous waits on us).  Instead,
+                        // we only check if the Task is completed.  Prior to that we will assume we are still loading.  Once this
+                        // task is completed, we know that the WaitUntilFullyLoadedAsync call will have actually finished and we're
+                        // fully loaded.
+                        var isFullyLoadedTask = workspaceStatusService?.IsFullyLoadedAsync(CancellationToken.None);
+                        var isFullyLoaded = isFullyLoadedTask is { IsCompleted: true } && isFullyLoadedTask.GetAwaiter().GetResult();
```

Comment:
does this .IsCompleted check actually work well?  i'm worried that some part of the IsFullyLaodedAsync call is actually async (Even trivially), and we just get a task back that this is always false for.  

i think i woudl like it if IsFullyLaodedAsync actually guaranteed the behavior you're looking for here by having a single task that it creates and caches and then returns to all future callers.  since i think we can only fully load once... that seems viable.

Reply:
@CyrusNajmabadi I agree the pattern isn't good, it's not new here. @sharwell might recall if we actually know if this works?

Reply:
Original conversation where this pattern was discussed: https://github.com/dotnet/roslyn/pull/48215#discussion_r514288546

There is probably a question as to whether we need to maintain the telemetry here at all, given everything that has gone on since Paul originally requested it.